src/activate/activate.c

```diff
@@ -321,13 +321,13 @@ static void sigchld_hdl(int sig, siginfo_t *t, void *data) {
 
         log_info("Child %d died with code %d", t->si_pid, t->si_status);
 
-        /* Wait for a dead child. */
-        (void) waitpid(t->si_pid, NULL, 0);
+        while (waitpid(-1, NULL, WNOHANG) > 0)
+                continue;
 }
```

Comment:
Hmm, the code is pretty broken actually. We should not use the siginfo_t passed in via the signal handler, as this function will be called only once if multiple SIGCHLD take place at the same time. The siginfo_t will then only contain information about the most recent signal.

Instead, we should use waitid() in a loop and print the information acquired from it instead.